src/utils/stacUtils.tsx

```diff
@@ -117,3 +134,14 @@ export const getFormattedSTACDateStr = (dates: ExtractedDates): string => {
   if (!dates.start && dates.end) date = `Up to ${dates.end}`;
   return date;
 };
+
+export const addLicenceLink = (collection: Collection): string => {
```

Comment:
An easier way to achieve this is:
```
export const addLicenceLink = (collection: Collection): string => {
  const licenceLink = collection.links.find(link => link.rel === 'licence');
  return licenceLink ? licenceLink.href : '';
};
```

Reply:
Watch out that there can be more than one licence link with different MIME types - the SPDX hosting story in particular is going to add two links, one an HTML fragment and one text. Depending on how the display is implement it most likely makes sense to choose a preferred one.

Reply:
Annoyingly, https://dev.eodatahub.org.uk/api/catalogue/stac/catalogs/supported-datasets/collections/cop-dem-glo-90 seems to have a licence link with no MIME type specified but it's a pdf:

```
    {
      "rel": "license",
      "href": "https://dev.eodatahub.org.uk/documents/20123/121286/CSCDA_ESA_Mission-specific+Annex_31_Oct_22.pdf",
      "title": "Copernicus DEM License"
    },
```

It's not currently looking promising with the ones we generate, either (this is for CMIP6), but at least we can get the `type` field added there:

```
       {
          "rel": "licence",
          "href": "https://dev.eodatahub.org.uk/harvested/default/spdx/license-list-data/main/html/0BSD.html"
        },
        {
          "rel": "licence",
          "href": "https://dev.eodatahub.org.uk/harvested/default/spdx/license-list-data/main/text/0BSD.txt"
        }
      ],
```

Reply:
@shahidraza-telespazio I should probably tag you here to make sure you know about the licence links above. The relation type has to be spelt `license` and there should be a `type` field - but maybe they were generated by incomplete code and haven't been regenerated yet?

Reply:
In the latest commit I've told it to return the .html one if it exists and to return the first one it finds if not. What do you think?

Reply:
I will also suggest DOMPurify as an alternative to iframing. Should be easy enough to include in the function (and others by the sounds of it as XSS shouldnt be underestimated considering users can upload their own data to their workspace catalog). Iframing can lead to mayhem with stylings and if DOMPurify achieves the same result I'd rather this.
https://github.com/cure53/DOMPurify

Reply:
I agree, DOMPurity looks like a good option.

Reply:
Actually, EODHP-991 should solve the sanitization problem. That will copy licences and sanitize them before hosting them at `/api/catalogue/licences/<workspace>/...` and pointing the link there.

Some extra sanitization or an iframe may still be worth it, though, in case any unsanitized links creep into our catalogue (but with the current implementation, with plain links, there's no XSS risk if those links point to another domain). Maybe we can save that until we're doing more work on how the licence and dataset details in general are displayed. It might be that we want to move the licence display to the dataset details page and use a lightbox for it, for example - perhaps Antoine will have some advice on that when the designs are more complete.

So, nothing to do here for now.